accounting_reporting_core/src/main/java/org/cardanofoundation/lob/app/accounting_reporting_core/resource/presentation_layer_service/ExtractionItemService.java

```diff
@@ -0,0 +1,73 @@
+package org.cardanofoundation.lob.app.accounting_reporting_core.resource.presentation_layer_service;
+
+import static java.math.BigDecimal.ZERO;
+
+import java.time.LocalDate;
+import java.util.List;
+import java.util.stream.Collectors;
+
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+import org.cardanofoundation.lob.app.accounting_reporting_core.domain.core.reconcilation.Reconcilation;
+import org.cardanofoundation.lob.app.accounting_reporting_core.domain.core.reconcilation.ReconcilationCode;
+import org.cardanofoundation.lob.app.accounting_reporting_core.domain.entity.*;
+import org.cardanofoundation.lob.app.accounting_reporting_core.repository.TransactionItemExtractionRepository;
+import org.cardanofoundation.lob.app.accounting_reporting_core.resource.views.ExtractionTransactionItemView;
+import org.cardanofoundation.lob.app.accounting_reporting_core.resource.views.ExtractionTransactionView;
+
+@Service
+@Slf4j
+@RequiredArgsConstructor
+@Transactional
```

Comment:
this will be automatically for reading and writing, I suggest you put explicitly that this is for reading and for writing only methods you need.

```
@Transactional(readOnly = true)
```

Reply:
Looking more into the code, you do not need any write operations, so just put what I wrote above, it has some hints to JPA to optimise queries.